### PR TITLE
[Internal Component] Convert environment version to str

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
When the environment version in the v1.5 component spec is a number, it will be considered as an int type when loading the component. This will cause an error in validation.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/internal/unittests/test_component.py [hunk 1]
```diff
@@ -334,9 +334,6 @@ def test_environment_dependencies_resolve(self, yaml_path: str, expected_dict: D
         component.environment.resolve(component._base_path)
         rest_obj = component._to_rest_object()
         assert rest_obj.properties.component_spec["environment"] == expected_dict
-        assert component.environment._validate_docker_section(
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/internal/unittests/test_component.py:337]: why remove this part?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/environment.py [hunk 2]
```diff
@@ -19,3 +19,9 @@ class InternalEnvironmentSchema(PathAwareSchema):
     name = fields.Str()
     version = fields.Str()
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/environment.py:20]: why not use the core  schema VersionField?

Comment (lalala123123) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/environment.py]: Change to VersionField

File: sdk/ml/azure-ai-ml/tests/internal/unittests/test_component.py (hunk 1 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/tests/internal/unittests/test_component.py]: revert it

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**